runtime/commonMain/src/kotlinx/serialization/internal/SerialClassDescImpl.kt

```diff
@@ -13,47 +13,50 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-@file:Suppress("OPTIONAL_DECLARATION_USAGE_IN_NON_COMMON_SOURCE")
+@file:Suppress("OPTIONAL_DECLARATION_USAGE_IN_NON_COMMON_SOURCE", "UNUSED")
 package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
 import kotlinx.serialization.CompositeDecoder.Companion.UNKNOWN_NAME
+import kotlin.LazyThreadSafetyMode.*
 import kotlin.jvm.JvmOverloads
 
-class MissingDescriptorException(index: Int, origin: SerialDescriptor) :
-    SerializationException("Element descriptor at index $index has not been found in $origin")
-
 open class SerialClassDescImpl @JvmOverloads constructor(
     override val name: String,
     private val generatedSerializer: GeneratedSerializer<*>? = null
 ) : SerialDescriptor {
+    /*
+     * Unused methods are invoked by auto-generated plugin code
+     */
     override val kind: SerialKind get() = StructureKind.CLASS
+    override val elementsCount: Int get() = annotations.size
 
     private val names: MutableList<String> = ArrayList()
     private val annotations: MutableList<MutableList<Annotation>> = mutableListOf()
     private val classAnnotations: MutableList<Annotation> = mutableListOf()
-
     private var flags = BooleanArray(4)
 
     private val descriptors: MutableList<SerialDescriptor> = mutableListOf()
-
-    private var _indices: Map<String, Int>? = null
-    private val indices: Map<String, Int> by lazy { buildIndices() }
+    private val indices: Map<String, Int> by lazy(PUBLICATION) { buildIndices() }
 
     @JvmOverloads
-    fun addElement(name: String, isOptional: Boolean = false) {
+    public fun addElement(name: String, isOptional: Boolean = false) {
         names.add(name)
         val idx = names.size - 1
         ensureFlagsCapacity(idx)
         flags[idx] = isOptional
         annotations.add(mutableListOf())
     }
 
-    fun pushAnnotation(a: Annotation) {
+    public fun pushAnnotation(a: Annotation) {
         annotations.last().add(a)
     }
 
-    fun pushDescriptor(desc: SerialDescriptor) {
+    public fun pushClassAnnotation(a: Annotation) {
```

Comment:
Any reasons that these changes are in the patch set? I'm not saying the change isn't good, but especially marking things public (when they are by default) is unrelated.

Reply:
It's just a maintenance work that is much easier to do when you pass by.
We are generally ok with such changes within the team

Reply:
>but especially marking things public

To clarify, this is the code style we (libraries team) stick with: if something is `public`, then it has to have explicit `public` modifier (see also [KEEP-45](https://github.com/Kotlin/KEEP/issues/45))